In [20]:
# Loading datasets into R
library(tidyverse)
players <- read_csv("players.csv")
sessions <- read_csv("sessions.csv")

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [25]:
summary(players)

  experience        subscribe       hashedEmail         played_hours    
 Length:196         Mode :logical   Length:196         Min.   :  0.000  
 Class :character   FALSE:52        Class :character   1st Qu.:  0.000  
 Mode  :character   TRUE :144       Mode  :character   Median :  0.100  
                                                       Mean   :  5.846  
                                                       3rd Qu.:  0.600  
                                                       Max.   :223.100  
                                                                        
     name              gender               Age       
 Length:196         Length:196         Min.   : 9.00  
 Class :character   Class :character   1st Qu.:17.00  
 Mode  :character   Mode  :character   Median :19.00  
                                       Mean   :21.14  
                                       3rd Qu.:22.75  
                                       Max.   :58.00  
                               

**(1) Data Description:**
-

players.csv
- **Number of observations**: 196
- **Summary statistics (2 d.p.)**: 5.85 mean hours played???????
- **Number of variables**: 7
    - experience <chr> The experience level of the player (Amateur, Regular, Pro, Veteran)
    - subscribe <lgl> Whether players have subscribed to the newsletter (TRUE, FALSE)
    - hashedEmail <chr>	The player's email, hashed for privacy
    - played_hours <dbl> The number of hours the player has spent on the server
    - name <chr> The player's name
    - gender <chr> The gender of the player (Male, Female, Non-binary, Agender, Two-Spirited, Prefer not to say)
    - Age <dbl> The players age
- **Any issues in the data**: None??????????
- **How the data was collected**: Dr. Frank Wood collected data on player actions as they navigate in his MineCraft server.

sessions.csv
- **Number of observations**: 1535
- **Summary statistics (2 d.p.)**: ???????
- **Number of variables**: 5
    - hashedEmail <chr> The player's email, hashed for privacy
    - start_time <chr> The start date and start time (format: DD/MM/YYYY HH:MM)
    - end_time <chr> The end date and end time (format: DD/MM/YYYY HH:MM)
    - original_start_time <dbl> The original start time using scientific notation (6 d.p. e+12)
    - original_end_time <dbl> ????????
- **Any issues in the data**: The start_time and end_time columns are not tidy, since there are two values (data and time) in eadch cell.
- **How the data was collected**: Dr. Frank Wood collected data on player actions as they navigate in his MineCraft server.

**(2) Questions:**
-

I will be exploring Question 1: What player characteristics and behaviours are most predictive of subscribing to a game-related newsletter, and how do these features differ between various player types?

Specific question: Can [explanatory variable(s)] predict [response variable] in [dataset]?

**(3) Exploratory Data Analysis and Visualization**
-

In [12]:
# Loading datasets into R
library(tidyverse)
players <- read_csv("players.csv")
sessions <- read_csv("sessions.csv")

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [13]:
# Wrangling

In [14]:
# Compute the mean value for each quantitative variable in the players.csv data set. 
# Report the mean values in a table format.

In [15]:
# Make a few exploratory visualizations of the data to help you understand it.
# Use our visualization best practices to make high-quality plots 
# (make sure to include labels, titles, units of measurement, etc)
# Explain any insights you gain from these plots that are relevant to address your question